website_and_docs/content/documentation/grid/advanced_features/external_datastore_sessions.en.md

```diff
@@ -0,0 +1,272 @@
+
+In short we now have a consistency problem that is so closely associated with the notion of microservices.
+
+## The solution
+
+The simplest solution would be to ensure that all the sessions map instances use a common shared data store. And thus we have a need to externalise this data store which holds running sessions information.
+
+Thanksfully Grid4 is already geared up to support this use case. 
+You can use a data store that is: 
+
+* Backed by your favourite database (or) 
+* Backed by a Redis Cache system
+
+so that all the running sessions map instances use a single shared data store.
+
+## Setup
+
+* [Coursier](https://get-coursier.io/docs/cli-installation) - As a dependency resolver, so that we can download maven artifacts on the fly and make them available in our classpath
+* [Docker](https://docs.docker.com/engine/install/) - To manage our PostGreSQL/Redis docker containers.
+
+## Database backed Session Map
+
+For the sake of this illustration, we are going to work with PostGreSQL database.
+
+We will spin off a PostGreSQL database as a docker container using a docker compose file.
+
+### Steps
+
+You can skip this step if you already have a PostGreSQL database instance available at your disposal.
+
+* Create a sql file named `init.sql` with the below contents:
+
+```sql
+CREATE TABLE IF NOT EXISTS sessions_map(
+    session_ids varchar(256),
+    session_caps text,
+    session_uri varchar(256),
+    session_stereotype text,
+    session_start varchar(256)
+ );
+```
+
+* In the same directory as the `init.sql`, create a file named `docker-compose.yml` with its contents as below:
+
+```yaml
+version: '3.8'
+services:
+  db:
+    image: postgres:9.6-bullseye
+    restart: always
+    environment:
+      - POSTGRES_USER=seluser
+      - POSTGRES_PASSWORD=seluser
+      - POSTGRES_DB=selenium_sessions
+    ports:
+      - "5432:5432"
+    volumes:
+    - ./init.sql:/docker-entrypoint-initdb.d/init.sql
+```
+
+We can now start our database container by running:
+
+```bash
+docker-compose up -d
+```
+
+_So our database name is `selenium_sessions` with its username and password set to `seluser`_
+
+If you are working with an already running PostGreSQL DB instance, then you just need to create a database named `selenium_sessions` and the table `sessions_map` using the above mentioned SQL statement.
+
+* Create a Selenium configuration file named `sessions.toml` with the below contents:
+
+```toml
+[sessions]
+implementation = "org.openqa.selenium.grid.sessionmap.jdbc.JdbcBackedSessionMap"
+jdbc-url = "jdbc:postgresql://localhost:5432/selenium_sessions"
+jdbc-user = "seluser"
+jdbc-password = "seluser"
+```
+
+_Note:_ If you plan to use an existing PostGreSQL DB instance, then replace `localhost:5432` with the actual host and port number of your instance.
+
+* Below is a simple shell script (let's call it `distributed.sh`) that we will use to bring up our distributed grid.
+
+```bash
+SE_VERSION=4.6.0
```

Comment:
```suggestion
SE_VERSION=<current_selenium_version>
```

Reply:
I prefer to leave this placeholder because it makes the docs easier to maintain.

Reply:
Fixed